### PR TITLE
Fixed Auto-Crafter issue

### LINKED ISSUE #35: [BUG] UU-Matter recipes not work in Slimefun Auto-Crafter
## Description (REQUIRED)

UU-Matter recipes not work in Slimefun Auto-Crafter.

## Steps to reproduce the Issue (REQUIRED)

1. Give and build Slimefun Auto-Crafters system.
2. Right click with item that can be made of UU-Matter.
3. UU-Matter recipe not selectable.

## Expected behavior (REQUIRED)

UU-Matter recipes works in Slimefun Auto-Crafter.

## Server Log

none

## Environment (REQUIRED)

Tuinity git-Tuinity-"c303521" (MC: 1.16.5)
Slimefun DEV - 864 (git 82366191)
Metrics-Module #21
Java 11

Installed Addons: (5)
  InfinityExpansion vDEV - 100 (git 65892de0)
  LiteXpansion vDEV - 138 (git 004e3c61)
  SlimefunFix v3.0
  FoxyMachines vDEV - 42 (git ae8d8aba)
  DynaTech vDEV - 90 (git 9f27da61)

### PR DESCRIPTION
**Pull request by Laci444**
## Short Description
I fixed the issue with the Auto-Crafters.

## Additions/Changes/Removals
UU-Matter recipes can be used in Auto-Crafters.

## Related Issues

Resolves #35 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/uumatter/UUMatter.java [hunk 1]
```diff
@@ -6,31 +6,31 @@
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/uumatter/UUMatter.java:9]: Never star import

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
Im not the biggest fan of making them all slimefun items since they are just vanilla items @WalshyDev what do you think

File: src/main/java/dev/j3fftw/litexpansion/uumatter/UUMatter.java (hunk 1 above)
Comment (Laci444) [reply src/main/java/dev/j3fftw/litexpansion/uumatter/UUMatter.java:9]: Sorry it's done by IDE

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/uumatter/UUMatter.java:9]: It's fine that's why we review them

**Comment by J3fftw1** [comment]
Hey, Thanks for the pr i got some other things i do want to change @WalshyDev and I have been talking about adding our own auto crafter for LiteXpansion that work with the UU matter stuff. This crafter purpose will solely be crafting uu matter into other items.
But that might be out of the scope of the PR but thats up to you we could always change it later on. Thanks again for making the PR <3

**Comment by Laci444** [comment]
> Im not the biggest fan of making them all slimefun items since they are just vanilla items @WalshyDev what do you think

This is the only solution I found without changing the Auto-Crafters or adding new ones.

**Comment by J3fftw1** [comment]
> 
> 
> > Im not the biggest fan of making them all slimefun items since they are just vanilla items @WalshyDev what do you think
> 
> This is the only solution I found without changing the Auto-Crafters or adding new ones.

I'm pretty sure if you do it this way the items aren't usable in vanilla crafting tables anymore since they aren't actually vanilla items anymore because you add a SF ID

**Comment by WalshyDev** [comment]
This PR has a few issues but the main one is you're now causing all these vanilla items to have an NBT tag which is redundant. This will cause stacking issues and issues with other plugins.

The best solution to this is our own crafter. The issue is due to these not being SF items (which we don't want them to be) but that's required for the enhanced crafter. Our own table will check if there's an enhanced recipe for our vanilla item.

So, due to this PR not doing an appropriate fix I will be closing it, if you wish to PR with a new auto-crafter then you're free to do so.

**Pull request closed without merging**